packages/vite-plugin-cloudflare/src/miniflare-options.ts

```diff
@@ -204,41 +204,62 @@ export function getMiniflareOptions(
 				],
 				serviceBindings: {
 					...workerOptions.serviceBindings,
-					__VITE_FETCH_MODULE__: async (request) => {
-						const [moduleId, imported, options] = (await request.json()) as [
-							string,
-							string,
-							FetchFunctionOptions,
-						];
-
-						// For some reason we need this here for cloudflare built-ins (e.g. `cloudflare:workers`) but not for node built-ins (e.g. `node:path`)
-						// See https://github.com/flarelabs-net/vite-plugin-cloudflare/issues/46
-						if (moduleId.startsWith('cloudflare:')) {
-							const result = {
-								externalize: moduleId,
-								type: 'module',
-							} satisfies vite.FetchResult;
-
-							return new MiniflareResponse(JSON.stringify(result));
-						}
-
-						const devEnvironment = viteDevServer.environments[
-							workerOptions.name
-						] as CloudflareDevEnvironment;
+					__VITE_INVOKE__: async (request) => {
+						const payload = (await request.json()) as vite.CustomPayload;
+
+						invariant(
+							payload.type !== 'custom',
+							`unexpected payload type sent to the vite-invoke binding: "${payload.type}"`,
+						);
+
+						invariant(
+							payload.event !== 'vite:invoke',
+							`unexpected payload event sent to the vite-invoke binding: "${payload.event}"`,
+						);
+
+						const invokePayloadData = payload.data as {
+							id: string;
+							name: string;
+							data: [
+								string,
+								string,
+								FetchFunctionOptions, // TODO: this should be the invoke equivalent
```

Comment:
`FetchFunctionOptions` seems to be the correct type here actually! ([source](https://github.com/vitejs/vite/blob/584a573e1fdfa64018b2c56d147f2b4fec2d9733/packages/vite/src/shared/invokeMethods.ts#L83))